### PR TITLE
Add get support page

### PR DESCRIPTION
**Pull request by fpapon**
Proposal to add a `get support` page to describe how community and commercial support is working, like we have in many other Apache projects.

### CODE REVIEW

**Review by jbonofre — APPROVED** [review]

**Review by rmannibucau — APPROVED** [review]

**Comment by C0urante** [comment]
@jbonofre @rmannibucau did you actually review this PR? Part of it is copy+pasted with no changes from the [Apache Camel docs](https://github.com/apache/camel/blob/74b6e1994a33fe54c3f5a5ad2acc5849148dbb16/docs/user-manual/modules/ROOT/pages/commercial-camel-offerings.adoc#commercial-camel-offerings). I would hope that anyone who actually looked at the diff would have spotted that.

File: support.html
Comment (C0urante) [inline support.html]: Wrong project.

**Comment by jbonofre** [comment]
@C0urante yup it's a copy paste from Camel. I already asked @fpapon to update the page to mention Kafka instead of Camel.

File: support.html
Comment (jbonofre) [inline support.html]: It should be Kafka here.

**Review by jbonofre — CHANGES_REQUESTED** [review]

**Comment by rmannibucau** [comment]
@C0urante yes, this is a small typo @fpapon is aware of but the proposal follows ASF guidelines and the fix is trivial (before or after the merge) so a clear plus for Kafka community from the last year requests I saw, but you are right, we should have highlighted it more.

**Comment by fpapon** [comment]
@C0urante good catch! Sorry for the copy/paste error, it's ok now.

**Review by rmannibucau — APPROVED** [review]
All good now

side note FTR (unrelated to this PR but since I hit it for each review, the docker command does not work on a docker setup not outdated and) launch command needs to be replaced with `docker run --security-opt seccomp:unconfined --rm --name mypreview -p 8080:80 -v "$PWD":/usr/local/apache2/htdocs/:z kafka-site-preview`)

File: support.html
Comment (ableegoldman) [inline support.html]: I feel like this should link to JIRA directly, and also mention it by name instead of saying "one of our issue trackers" -- we really only have the one. The "contributing" link would be more appropriate for the following sentence, which does indeed appear to be about contributing rather than reporting issues or "community support". Frankly this section is kind of all over the place but I think it's fine as long as we hit on the important notes -- like where to report issues

File: support.html
Comment (ableegoldman) [inline support.html]: It might be good to be a bit more specific here. For example most support related questions should probably go to the `users@apache.kafka.org` mailing list specifically

File: support.html
Comment (ableegoldman) [inline support.html:36]: It seems very weird to me to add this page with only a single commercial support offering listed. Honestly it feels a bit inappropriate to use the AK site to list any commercial offerings at all, but I'm not sure what the official policy is there. If we do include a commercial support section then imo we should consider reaching out to some of the other major companies offering Kafka support to ask if they would like to be included so that we provide a more complete picture of the available options. Otherwise it feels like we (the Apache Kafka project) are advocating for a particular commercial offering, which we definitely are not and should not.

I still think the first part of this page could be very valuable (with the above suggestions) and would recommend splitting this PR up so that we can merge the community support section while discussing the suitability of the commercial support section

File: support.html
Comment (ableegoldman) [reply support.html]: Another common source of help is the Confluent Community Slack. Yes, it's run by Confluent, but it's free and is fairly active with people, both users and devs, from across a wide range of companies. Not sure if we'd need to get permission to include that or not.

I also see people ask and get good answers on StackOverflow, though generally speaking the users mailing list and Confluent Community Slack are the most active in my experience

**Review by ableegoldman — CHANGES_REQUESTED** [review]

File: support.html
Comment (rmannibucau) [reply support.html:36]: An alternative to reaching out (which feels like the opposite) is to add an edit link (github) and encourage people to pr to add themselves.
Will make it more open and less closed to a single company so cna be a trade off maybe.
Wdyt?

File: support.html
Comment (fpapon) [reply support.html]: There is an existing ASF Slack and a Kafka channel so why not using it?

Comment (fpapon) [reply support.html]: I will mention and add a link to StackOverFlow.

File: support.html
Comment (fpapon) [reply support.html]: Ok, I will update to be more specific about the issue tracker.

File: support.html
Comment (fpapon) [reply support.html:36]: I added the first company because I don't know the other and like it's mention in the text, company are encouraged to add themselves by providing a PR. At the beginning we need to start with one company but if people want to add other companies in this PR with comments, I can add them to start with more than one before merging.

Comment (fpapon) [reply support.html:36]: @ableegoldman do you think we can start a thread on the mailing list to ask which company want to be add in the commercial list support?

Comment (showuon) [reply support.html:36]: I think you should start a discussion thread to ask if we "should" have the "Commercial support" section first. I agree with @ableegoldman that it's weird the Apache Kafka site contains any commercial content.

File: support.html
Comment (showuon) [reply support.html]: I know @ableegoldman is trying to list all helpful resources here, but I don't think adding `Confluent Community Slack` or `StackOverflow` is appropriate here. `Confluent Community Slack` is run by Confluent, so it is not a neutral recourse. About the `StackOverflow`, I think if we list `StackOverflow` here, does that imply `users@apache.kafka.org` or `dev@apache.kafka.org` are not useful? I think, as a community, we should recommend users to ask/answer questions in the community, instead of in the 3rd party site.

**Review by showuon — COMMENTED** [review]
@fpapon , thanks for the PR. Could you explain the motivation that why we should add `Get Support` page? Why is the `Contact Us` page not enough? Thanks.

File: support.html
Comment (rmannibucau) [reply support.html:36]: For one rational (coming from other ASF projects): asking for help always require some sort of registration and when a company just want support (often SLA related or expertise related) this is perceived as negative so several ASF projects (most of the ones able to run as standalone, ie not libs) started to add such a page or at least point to a page enabling to find out the info.

hope it helps

**Comment by fpapon** [comment]
@showuon There is a lot of ASF projects that provide a page with community support, it help the adoption and the growth of the project because the users need commercial support for production (SLAs), consulting or training. So if some company are listed in the project website it can be very useful for them to find support. You can check Camel, ActiveMQ, Karaf websites for example.

**Comment by fpapon** [comment]
> @fpapon , thanks for the PR. Could you explain the motivation that why we should add `Get Support` page? Why is the `Contact Us` page not enough? Thanks.

- This page explain more how to get supported. @ableegoldman didn't said that it's weird to have commercial support list, but it's weird to have only one company listed, that I'm agree and we can ask on the mailing list if some other company want to be listed.

- Agree for the vendor neutral slack channel, that's why I proposed to list the existing official ASF Kafka slack channel. About StackOverFlow, as there is a lot of content on that channel, it can be used as a FAQ for the user but users can also ask on the mailing list. I think users can search content more easily on StackOverFlow rather than on the mailing list thread history.

**Comment by mimaison** [comment]
You have to admit this PR raised quite a few red flags!

First it's directly copied from Camel without replacing mentions to Kafka. Then within minutes it's approved by 2 other ASF members clearly without looking at the diff. Finally it's adding your company as the sole provider for commercial support.

A more open approach would have been to reach out the Kafka community on the dev or users lists and discuss a nice way to build such a page. I tend to agree that a page listing commercial offerings is helpful but I'm sure you understand why we can't merge the PR as is.

**Comment by rmannibucau** [comment]
@mimaison well I wouldn't say red flags but I fully understand the surprise on kafka side if it was never discussed on/offline first - we had the same debate on TomEE side years ago.
But it is not uncommon at apache to have such a page, even with a single company or even with no company and request for companies:

* https://hop.apache.org/community/commercial/ (0)
* https://struts.apache.org/commercial-support.html (1)
* https://directory.apache.org/commercial-support.html (1)
* https://superset.apache.org/community (direct link to the company behind - this one is more surprising for me since instead of encouraging the "registration" of vendors it hides it somehow but guess it is a small clumsiness)
* https://tomee.apache.org/commercial-support.html (2)
* https://plc4x.apache.org/users/commercial-support.html (2)
* https://camel.apache.org/community/support/
* https://openmeetings.apache.org/commercial-support.html
* https://guacamole.apache.org/support/
* https://cwiki.apache.org/confluence/display/HADOOP2/Distributions+and+Commercial+Support
* https://activemq.apache.org/support
* https://netbeans.apache.org/front/main/help/commercial-support/
* https://royale.apache.org/royale-commercial-support/
* (way more but guess you got the idea)

To give an example of the "rationale behind" please have a look to https://github.com/apache/superset/issues/8852 issue.

An important point to take into consideration is that several products - and I think Kafka is exactly there - wouldn't live without external support (it is true for most broker or server) so it is guaranteeing its community and helping its user base to do this kind of page - and trust me, ~5 years ago I didn't understand that as well as today so I say it very humbly.

So overall, due to kafka adoption it can only be good to get such a page IMHO but I agree that if you feel it is not straight forward and it must be discussed more deeply cause Kafka wants to write its own content, a thread sounds the way to move it forward and a core commiter should then probably take the leadership on that "doc" track and this pr marked as pending while this work is done.

**Comment by mimaison** [comment]
As I said, I think a page listing commercial offerings is useful. I was just commenting on the approach you took.

I don't think we necessarily need a committer to build this page (obviously a committer will have to review the PR before merging). If you want you can start the discussion.

**Comment by fpapon** [comment]
> You have to admit this PR raised quite a few red flags!
> 
> First it's directly copied from Camel without replacing mentions to Kafka. Then within minutes it's approved by 2 other ASF members clearly without looking at the diff. Finally it's adding your company as the sole provider for commercial support.
> 
> A more open approach would have been to reach out the Kafka community on the dev or users lists and discuss a nice way to build such a page. I tend to agree that a page listing commercial offerings is helpful but I'm sure you understand why we can't merge the PR as is.

Hi @mimaison,

I admitted that I made a mistake about copy/paste from Camel website and I fixed it. About adding my company, as I explained in other comments, it's just a starting point to list companies. I cannot add the others for trademark purpose because I'm not owning them but everybody can add their company as explain in the text or I'm open to do it if people ask me in this PR.

About the approach, I already start a thread on the mailing list about this topic before doing a proposal, here the link to thread:
https://lists.apache.org/thread/kkox33rhtjcdr5zztq3lzj7c5s7k9wsr

Sorry, it was last year, so may be I would have to resend it again?

**Comment by fpapon** [comment]
@mimaison I replied to the original thread on the mailing list to rebirth the discussion.

**Comment by jbonofre** [comment]
@mimaison I take the blame on me: I did mistakes with the bright side of things. 
1. I thought this kind of change was approved by the community/PMC
2. As the same page exists in several ASF projects, and it was a copy from Camel, I thought it was ok straight forward without checking the diff in details.

I share your points, but I would not be too sharp. @fpapon is just trying to help. I would add more companies/providers in this page as part of this PR. I think we can say that it's for the good of the Kafka community and diversity.

I'm happy to chat with you about that on Slack or on a call if you want.

**Comment by mimaison** [comment]
No worries, there's no need to blame anybody for such a small issue. The initiative to add this page is, in my opinion, good and contributions are always welcome. It's just the initial content and quick approvals that looked concerning.

@fpapon As pointed by several committers there are a few issues with the current content but we should be able to agree on what to do and update this PR accordingly. I wonder if the first step would be to decide whether we want this page on the website. It's probably best to email the dev to get some agreement, or at least allow people to raise objections and then start gathering content.

**Comment by fpapon** [comment]
@mimaison ok thanks for your feeedbacks. I will update the PR according to the comments of the people and then resend an email to the mailing for the proposal and see how the community react on it.

**Comment by fpapon** [comment]
@ableegoldman I pushed some changes, feel free to review/comment :)

**Comment by fpapon** [comment]
As the Apache Kafka PMC rejeted this proposal on the mailing list, I'm closing this PR.
Link to the thread:
https://lists.apache.org/thread/sld85ly7fvvvlb5bh2856qqcpnckg2on

**Comment by jbonofre** [comment]
@fpapon I guess you mean Kafka PMC :) Not Karaf :)

**Comment by fpapon** [comment]
> @fpapon I guess you mean Kafka PMC :) Not Karaf :)

Ah yes, sorry, split brain :)
I fixed.

**Pull request closed without merging**
